### PR TITLE
Dynamically adjust the compatibility notice in the latest docs

### PR DESCRIPTION
**Pull request by YuriSizov**
Closes https://github.com/godotengine/godot-docs/issues/6452.

This PR makes the compatibility notice dynamic by checking if the stable version of the document exists using a HEAD request. If it does, the old message is displayed (I've adjusted it a bit to better match the new ones). If it doesn't exist, then this fact is stated instead, with a link to do a search in the stable documentation using this page's title.

Code feels rather hacky, but it's probably no worse than what we already do here.

Here's how it looks in practice. _Note, that it should be faster and less noticeable than that in reality. I've throttled my networking speed to better showcase different states._

https://user-images.githubusercontent.com/11782833/205526824-73deabac-1a04-4818-8032-0f461ac7e972.mp4

https://user-images.githubusercontent.com/11782833/205526826-4ee93eef-bca1-42a1-b4b0-98de1c877d0e.mp4

### CODE REVIEW

**Review by mhilbrunner — APPROVED** [review]
Code LGTM, lets see what Calinou thinks.

How does this interact with localization? Can this message be localized? Can the current one?
I've got no idea :)

File: _static/js/custom.js [hunk 1]
```diff
@@ -253,22 +253,63 @@ $(document).ready(() => {
...
     // Add a compatibility notice using JavaScript so it doesn't end up in the
     // automatically generated `meta description` tag.
-    const strippedUrl = [location.protocol, '//', location.host, location.pathname].join('');
-    const updatedUrl = strippedUrl.replace('/latest/', '/stable/');
+
+    const baseUrl = [location.protocol, '//', location.host, location.pathname].join('');
+    // These lines only work as expected in the production environment, can't test this locally.
+    const fallbackUrl = baseUrl.replace('/latest/', '/stable/');
+    const homeUrl = baseUrl.split('/latest/')[0] + '/stable/';
+    const searchUrl = homeUrl + 'search.html?q='
```
Comment (Calinou) [inline _static/js/custom.js:261]: Missing semicolon:

```suggestion
    const searchUrl = homeUrl + 'search.html?q=';
```

**Review by Calinou — APPROVED** [review]
Should be good to merge after applying the suggestion.

**Comment by YuriSizov** [comment]
Done!

**Comment by mhilbrunner** [comment]
Merged. Let's see if this works out... thanks! :)

**Pull request merged**